### PR TITLE
Save to file & tests

### PR DESCRIPTION
**Pull request by agabylicka**
Added writing to file and tests.

### CODE REVIEW

File: src/main/java/org/CharacterCreator/DataModel/Character.java [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import java.util.List;
 
 public class Character {
-    private Race race;
+    private String name;
+    private String race;
     private String backstory;
     private List<Feature> features;
-    private CharacterClass characterClass;
+    private String characterClass;
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/Character.java]: Why is it a String?

File: src/main/java/org/CharacterCreator/DataModel/Character.java [hunk 2]
```diff
@@ -1,12 +1,13 @@
-package org.example;
+package org.CharacterCreator.DataModel;
 
 import java.util.List;
 
 public class Character {
-    private Race race;
+    private String name;
+    private String race;
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/Character.java]: Why is race a String?

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import java.io.File;
+import java.io.IOException;
+
+public class FileWriter {
+     public String writeFile(String name, org.CharacterCreator.DataModel.Character character, String selection, String background) {
+        File file = new File("src\\main\\java\\org\\CharacterCreator\\" + name + ".txt");
+
+         java.io.FileWriter writer = null;
+         BufferedWriter bufferedWriter = null;
+        try {
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Please use try with resources

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package org.CharacterCreator.DataModel;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+
+public class FileWriter {
+     public String writeFile(String name, org.CharacterCreator.DataModel.Character character, String selection, String background) {
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Race and class names can be passed on from the main method

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
+package org.CharacterCreator.DataModel;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+
+public class FileWriter {
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Please change the name of the file

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 4 above)
Comment (blazejprzyluski) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: org.CharacterCreator.DataModel. check why this import is there

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+     public String writeFile(String name, org.CharacterCreator.DataModel.Character character, String selection, String background) {
+        File file = new File("src\\main\\java\\org\\CharacterCreator\\" + name + ".txt");
+
+         java.io.FileWriter writer = null;
+         BufferedWriter bufferedWriter = null;
+        try {
+            writer = new java.io.FileWriter(file); bufferedWriter = new BufferedWriter(writer);
+            bufferedWriter.flush();
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: to be deleted

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+            bufferedWriter.write("Name: " + character.getName());
+            bufferedWriter.newLine();
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Please decide on the approach of creating new lines

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+            ioException.printStackTrace();
+        } finally {
+            try {
+                bufferedWriter.close();
+                writer.close();
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+        return null;
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Why returning null?

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public boolean removeFile(String name) {
+         File file = null;
+         try { file = new File("src\\main\\java\\org\\CharacterCreator\\" + name + ".txt");
+            file.delete();
+        } catch (Exception e) {
+            return false;
+        }
+        return true;
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: return the result of file.delete()

File: src/main/java/org/CharacterCreator/HTTP/AllSpellsResult.java [hunk 10]
```diff
@@ -0,0 +1,34 @@
+package org.CharacterCreator.HTTP;
+
+public class AllSpellsResult {
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/HTTP/AllSpellsResult.java]: This class is redundant

File: src/main/java/org/CharacterCreator/Main.java [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+            if (raceRespond.contains(race)) {
+                existingRace = true;
+                character.setRace(race);
+            }
+        } while (!existingRace);
+
+        boolean existingCLass = false;
+        String className;
+
+        System.out.println("All races are: barbarian, bard, cleric, druid, fighter, monk, paladin, ranger, rogue, sorcerer, warlock, wizard");
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/Main.java]: All classes*

File: src/main/java/org/CharacterCreator/Main.java [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+    }
+}
+
+
+
+
+
+
+
+
```
Comment (blazejprzyluski) [inline src/main/java/org/CharacterCreator/Main.java]: Unnecessary empty space!

File: src/test/java/SpellsMAPPERTest.java [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+        Assertions.assertNotNull(actual);
+        Assertions.assertFalse(actual.getName().isEmpty());
+
+    }
+
+    public String readFile(String fileName) {
+        StringBuilder result = new StringBuilder();
+        BufferedReader reader = null;
+
+        try {
```
Comment (blazejprzyluski) [inline src/test/java/SpellsMAPPERTest.java]: Please use try with resources

File: src/test/java/SpellsMAPPERTest.java [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+        // when
+        Spells actual = SpellsMAPPER.convertJSON(json);
+
+        // then
+        Assertions.assertNotNull(actual);
+        Assertions.assertFalse(actual.getName().isEmpty());
+
+    }
+
+    public String readFile(String fileName) {
```
Comment (blazejprzyluski) [inline src/test/java/SpellsMAPPERTest.java]: Please make private

File: src/test/java/TestClassInformation.java [hunk 15]
```diff
@@ -28,9 +28,14 @@ public void testReadAllClassesInformation() {
 
     @Test
     public void testNullClassInformation() {
+        // given
         String index = null;
         ClassInformation classInformation = new ClassInformation();
-        Assertions.assertThrows(RuntimeException.class, () -> classInformation.getClassInformation(index));
+        // when
+        var actual = classInformation.getClassInformation(index);
```
Comment (blazejprzyluski) [inline src/test/java/TestClassInformation.java]: Why var?

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 5 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Done

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 6 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Done

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 7 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Done

File: src/main/java/org/CharacterCreator/Main.java (hunk 12 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/Main.java]: Done

File: src/test/java/TestClassInformation.java (hunk 15 above)
Comment (agabylicka) [reply src/test/java/TestClassInformation.java]: Fixed

File: src/test/java/SpellsMAPPERTest.java (hunk 14 above)
Comment (agabylicka) [reply src/test/java/SpellsMAPPERTest.java]: Fixed

File: src/main/java/org/CharacterCreator/Main.java (hunk 11 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/Main.java]: Done

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 4 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Second fixed

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 3 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Done

File: src/test/java/SpellsMAPPERTest.java (hunk 13 above)
Comment (agabylicka) [reply src/test/java/SpellsMAPPERTest.java]: Fixed

File: src/main/java/org/CharacterCreator/HTTP/AllSpellsResult.java (hunk 10 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/HTTP/AllSpellsResult.java]: Fixed

File: src/main/java/org/CharacterCreator/DataModel/Character.java (hunk 1 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/Character.java]: Done

File: src/main/java/org/CharacterCreator/DataModel/Character.java (hunk 2 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/Character.java]: Done

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 4 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: First fixed

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 8 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Done

File: src/main/java/org/CharacterCreator/DataModel/FileWriter.java (hunk 9 above)
Comment (agabylicka) [reply src/main/java/org/CharacterCreator/DataModel/FileWriter.java]: Done

**Review by blazejprzyluski — APPROVED** [review]

**Pull request merged**